lib/super_digit.rb

```diff
@@ -1,15 +1,45 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n) one loop based on the arraylength which is based on the input
+# Space Complexity -O(n) since I make the digit_array and the call stack
 def super_digit(n)
+
+  digit_array = n.to_s.split('')
+
+  sum = 0
+
+  digit_array.each do |i|
+    sum += i.to_i
+  end
+
+  if sum <= 9
+    return sum
+  end
+
+  return super_digit(sum)
    
 end
   
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - o(n*k)
+# Space Complexity - same as above. There is another loop here but it would just make it O(2n) and we drop the constants
 def refined_super_digit(n, k)
```

Comment:
Think about a dynamic programming solution it can bring it down to O(n + k)

```ruby
def refined_super_digit(n, k)
  sum = super_digit(n)
  sum *= k
  return super_digit(sum)
end
```